hubblestack/extmods/fdg/stat.py

```diff
@@ -0,0 +1,264 @@
+
+1. Sample FDG profile, with inline comments:
+This profile makes direct use of stat module by passing filepath directly in params
+*****************************************************
+main:                             # start of this profile's main module
+    module: stat.check_stats      # this tells FDG to call stat submodule's check_stats function
+    args:                         # arguments to check_stats function
+        - filepath: /etc/docker/daemon.json
+          mode: 644
+          gid: 0
+          group: root
+          uid: 0
+          user: root
+          allow_more_strict: True
+          match_on_file_missing: True
+*****************************************************
+
+2. Sample FDG profile, with inline comments:
+This profile chains the stat module, so that some module can pass filepath to stat module.
+*****************************************************
+main:                           # start of this profile's main module
+    module: osquery.query       # this tells FDG to call osquery submodule's query function
+    args:
+       - 'select path as filepath from FILE where path="/etc/docker/daemon.json";'
+    xpipe:                      # This tells FDG to pass the output of the above query to check_stats functions of this profile
+       check_stats
+
+check_stats:                    # this profile's check_stats module
+    module: stat.check_stats   # this tells FDG to call stat submodule's check_stats function
+    args:
+        - mode: 644
+          gid: 0
+          group: root
+          uid: 0
+          user: root
+          allow_more_strict: True
+          match_on_file_missing: True
+
+*****************************************************
+
+If `match_on_file_missing` is omitted, success/failure will be determined
+entirely based on the other arguments. If it's set to True and
+the file is missing, then it will be considered a match (success).
+If it's set to False and the file is missing, then it
+will be considered a non-match (failure).
+If the file exists, this setting is ignored.
+'''
+
+import logging
+import salt.utils
+import salt.utils.platform
+import os
+log = logging.getLogger(__name__)
+
+
+def check_stats(params='', chained=None, chained_status=None):
+    '''
+
+    :param params: dictionary of parameters to match with the file stats
+    :param chained: file path can be passed as chained
+    :param chained_status: Status returned by the chained function.
+    :return: tuple with (status(Boolean), result(dict))
+    This function takes into input stat params that are to be matched with stats of given file.
+    The filepath can be provided either directly in params or through chaining. See example above.
+    '''
+
+    if params == '' or params is None:
+        ret = {'Failure' : 'invalid input, no params provided'}
+        return False, ret
+
+    if chained:
+        log.info("value of 'chained' is not null, using {0} value as filepath".format(chained))
```

Comment:
Do not use format before passing to log.info ... log.info already uses the older style formatting rules so this is somewhat redundant and the interpolation may not be necessary.

`log.info("value of 'chained' is not null, using %s value as filepath", chained)`

Reply:
Done.